added pg_proctab functions

davecramer opened the pull request:
This is the easy way.  Just copied the code from pg_proctab in and modified the makefile. Future work to make it more like pgnodemx to use the parsing functions in pgnodemx

davecramer committed:
added pg_proctab functions

davecramer committed:
mostly working with pgnodemx parsing still a WIP

jconway commented on parseutils.c:
```diff
@@ -542,6 +542,8 @@ parse_keqv_line(char *line)
 	return values;
 }
 
+
```
gratuitous white-space addition

davecramer replied:
ya, will deal with this, thanks for reviewing

jconway commented on pg_proctab.c:
```diff
@@ -16,20 +16,54 @@
 #include <sys/stat.h>
```
Admittedly pgnodemx.c is getting a bit too long, but in any case the other "proc" functions are all defined in there. I think for this first addition of "proctab" functions we should keep everything in pgnodemx.c for the sake of consistency. A later cleanup might be good which moves all of the "proc" functions to their own source file (perhaps procfuncs.c?).

davecramer replied:
Fair enough. Once it works I can move it back into pgnodemx.c

davecramer committed:
fix whitespace

davecramer committed:
remove () from comm

davecramer committed:
get the uid and username

davecramer committed:
sorted out cpu_time

davecramer committed:
removed pretty much all of the original pg_proctab code, extra functions add pg_loadavg

davecramer committed:
added copyright

davecramer committed:
renamed file to procfunc updated Makefile

davecramer committed:
renamed functions to be consistent

jconway committed:
Add pg_proctab equivalent functionality

Add procfs functions that expose the same basic information
as pg_proctab, but do it in a way consistent with the rest
of pgnodemx. Also provide wrapper compatibility functions
with the hope that pgnodemx can serve as a remote for pg_top.
So far this is untested.

This commit lacks proper documentation, but it is getting
unweildy and is feature complete I believe, so look for the
missing docs in a later commit.

davecramer commented on genutils.c:
```diff
@@ -745,3 +747,19 @@ pg_ulltoa_n(uint64 value, char *a)
+{
+	Numeric		num1 = PG_GETARG_NUMERIC(0);
+	Numeric		num2  = int64_to_numeric(sysconf(_SC_PAGESIZE));
+	Numeric		res;
```
too many spaces. tab issues ?

davecramer commented on parseutils.c:
```diff
@@ -623,3 +623,43 @@ parse_space_sep_val_file(char *ftr, int *nvals)
+		int		nrow = *nlines;
+		int		ncol = 2;
+		int		i;
+
```
similar comment to spacing ?
